### ISSUE TITLE
app.element.io fails to load on browser startup

### ISSUE THREAD

**Issue by michaelkaye** [vector-im/element-web#16020]
### Description

Describe here the problem that you are experiencing, or the feature you are requesting.

Unable to load
Element can't load
Something went wrong and element was unable to load 

![unable_to_load](https://user-images.githubusercontent.com/1917473/102872650-e8874000-4437-11eb-9f8a-2b21d79317c9.png)

### Steps to reproduce

Ensure app.element.io opens by default on launching Firefox (for example, pin the tab, or have it open when closing the browser so it is automatically reloaded on start) - this has happened multiple times, but generally I only see it when firefox has been restarted at the start of a new day. Otherwise loading is perfectly fine

View the page

Describe how what happens differs from what you expected.

I see the screen shot above, not the UI

As an odd side effect, attempting to click on the link does not take me to element.io either.

Logs are hard because this is before any of the JS has loaded, but the console did display:

```
"09:12:32.596 Loading failed for the <script> with source “https://app.element.io/bundles/504aa24e021ebeda5b73/init.js”. app.element.io:1:1
09:12:32.597 ChunkLoadError: Loading chunk 12 failed. (error: https://app.element.io/bundles/504aa24e021ebeda5b73/init.js)     Webpack (6)
         e
         64
         64
         a        
        <anonymous>         
        <anonymous> 
index.ts:213:12     64 index.ts:213
webpack://src/vector/index.ts
```

Not sure if it's useful, but when i do refresh the page, it does go and download  `https://app.element.io/bundles/504aa24e021ebeda5b73/init.js` from the webserver, so the bundle version number is right

The network was up by the time i started firefox.

Prepared to believe it's a firefox bug, but reporting anyway.

### Version information

- **Platform**: web (in-browser)

For the web app:

- **Browser**: Firefox 83.0 (ad blocker installed, but 0 blocked on the page)
- **OS**:Ubuntu
- **URL**: app.element.io

**Comment by jryans** [comment]
@michaelkaye Is this a new issue, or has it been like this for a long time?

At first glance, it seems like it may be hard to fix this, as we rely on the browser to load the various scripts, and can't do much if they fail...

**Comment by michaelkaye** [comment]
@jryans i've noticed it in the last ... couple of weeks there's maybe been three examples? It's new, but that could also be new-firefox-new

Maybe one suggestion in the short term is a bit of TLC for this page, possibly with the simple suggestion of adding "Try refreshing this page" as the explicit action that will probably fix this sort of thing?

**Comment by jryans** [comment]
Sure, I'd be happy to accept a PR that clarifies this text to add that refreshing might help.

**Comment by lampholder** [comment]
I've also seen this a few times recently - quite reliably just now I've achieved it by going to https://develop.element.io/#/room/#testclassroom:matrix.org in a new private firefox window. It goes away after a few refreshes.

**Comment by lampholder** [comment]
Oh wait, I realise this is issue is _on browser startup_ - I'm seeing mine when I navigate to that page in a fresh browser tab, which is a bit different.

**Comment by jryans** [comment]
> I've also seen this a few times recently - quite reliably just now I've achieved it by going to https://develop.element.io/#/room/#testclassroom:matrix.org in a new private firefox window. It goes away after a few refreshes.

I can't duplicate this personally... it loads every time for me. In any case, it sounds like a different issue indeed. Let's keep this one scoped to browser startup.

**Comment by michaelkaye** [comment]
New firefox / new element-web, performing the above replication steps now gets me to a different outcome:

I now get this on the console:

```
12:19:22.792 Initialised rageshake. KeyboardShortcuts.tsx:271:22
12:19:22.793 To fix line numbers in Chrome: Meatball menu → Settings → Blackboxing → Add /rageshake\.js$ KeyboardShortcuts.tsx:271:22
12:19:22.795 Using Web platform KeyboardShortcuts.tsx:271:22
12:19:22.847 Uncaught (in promise) TypeError: NetworkError when attempting to fetch resource.
12:19:22.906 returning explicit system theme 2 KeyboardShortcuts.tsx:271:22
12:19:22.954 Loading skin... KeyboardShortcuts.tsx:271:22
12:19:23.363 Skin loaded! KeyboardShortcuts.tsx:271:22
12:19:23.397 set language to en-us KeyboardShortcuts.tsx:271:22
12:19:23.653 Cannot play media. No decoders for requested formats: audio/ogg, audio/ogg, audio/ogg, audio/ogg, audio/ogg, application/ogg, application/ogg, application/ogg, application/ogg, application/ogg app.element.io
12:19:32.635 The resource at “https://app.element.io/bundles/c58aafdb3f883a5dace0/element-web-app.js” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly.
```

And a blank page visible - no text errors.

**Comment by jryans** [comment]
I wonder if it's related to preloading the scripts... Given the message `The resource at “https://app.element.io/bundles/c58aafdb3f883a5dace0/element-web-app.js” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly`, I wonder if the browser gets confused after this point... still that would be a browser bug in my opinion.

**Comment by silasdb** [comment]
Hi. Since last update I'm having the same issue. Console gives me slight different error though:

    ChunkLoadError: "Loading chunk 22 failed.
    (missing: https://app.element.io/bundles/c6169dd06310247a46be/vendors~init.js)"

Strangely, I can open the js file if I copy the URL and paste it in the address bar.

I tried to clean all caches and cookies and even tried to open a different browser profile, with no success.

Using Firefox 66.0.1 (64-bit) (maybe too old?) in NetBSD 8.0.

**Comment by michaelkaye** [comment]
interestingly it's spreading: just now i came back to grafana and it had this error in the console and the generic loading failed screen:

```
 The resource at “https://grafana/public/fonts/roboto/RxZJdnzeo3R5zSexge8UUVtXRa8TVwTICgirnJhmVJw.woff2” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly. filesystem-space-available
 ```
 More tempted to say this is a firefox version-ism than anything else.

**Comment by t3chguy** [comment]
That's a symptom rather than the underlying issue. It sounds like Grafana just uses preloads a little too haphazardly, they are meant to be resources you will definitely be loading as part of starting the app. In the case of Element the reason that warning is shown is because app load fails and thus the browser gets upset that we asked it to preload stuff that we didn't end up needing

**Comment by TheCycoONE** [comment]
My wife has the same error as @silasdb on firefox 74 in Linux. It happens every time, with or without cache / private browsing. I haven't found any work around.

```
ChunkLoadError: "Loading chunk 22 failed.
(missing: https://app.element.io/bundles/c6169dd06310247a46be/vendors~init.js)"
```

**Comment by jryans** [comment]
> There is also `SyntaxError: invalid regular expression flag s` which is a feature added in Firefox 78; for for the two of us at least the problem is using an unsupported browser. The error message could be much better.

About this part, I've [added a check](https://github.com/vector-im/element-web/pull/16408) for this, so we'll soon show a nicer message if this is missing.

**Comment by silasdb** [comment]
Hi. Indeed I also get a `SyntaxError: invalid regular expression flag s`. Sorry for missing that in my previous message.

**Comment by jryans** [comment]
Okay, thanks for the follow up. We should have a new release next week with the check for regular expression support, so we can see how that changes things here (if at all). It does mean that that anyone seeing that regular expression error will soon be told their browser is unsupported though, as we only support the last 2 major versions of desktop browsers.

**Comment by silasdb** [comment]
On Wed, Feb 10, 2021 at 05:08:42AM -0800, J. Ryan Stinnett wrote:
> ... as we only support the last 2 major versions of desktop browsers.

Thanks for this. BTW, many services actually support one (two? three?) more older version but add a warning about it being deprecated.  I really got surprised that element-web stopped for me without previous warning.  I know that it requires more work but it would be nice to warn users so they get a chance to update their browsers.  Do you see a chance that it could be done for element-web?  Maybe have a warning for, say, only two weeks, before dropping support for the older version?

**Comment by t3chguy** [comment]
> told their browser is unsupported though

> add a warning about it being deprecated.

You will get an option to proceed at your own risk, where some features just might not work.

>  before dropping support for the older version?

The new features are already in use, the only thing that's being added is a deterministic warning to tell people their browser may lead to errors due to not having required features

**Comment by novocaine** [comment]
Closing this issue as we have made the error checking slightly nicer and the remaining issues seem to be related to unsupported browser versions.
